wrapanapi/systems/openstack.py

```diff
@@ -141,6 +141,16 @@ def ip(self):
                 if nic['OS-EXT-IPS:type'] == 'floating':
                     return str(nic['addr'])
 
+    @property
+    def all_ips(self):
+        """ Get all the IPs on the machine
+
+        Returns: (list) the addresses assigned to the machine
+        """
+        networks = self._get_networks()
+        # pull list of nics for each network, and addr from each nic
+        return [str(nic['addr']) for nics in networks.values() for nic in nics]
```

Comment:
Should we not check `networks.values()` and `nics` for being `NoneType`
We might get `TypeError: 'NoneType' object is not iterable`

Reply:
Its possible, I'm not finding a clear answer in the openstack novaclient v2 source for a guaranteed type on the `addresses` list in info, but I'm pretty sure it'll always be a dict. In which case `{}.values()` is just empty and you get an empty list out of this.

Either way I'm changing it up completely, to use `novaclient.v2.servers.Server.networks` to build the list in the same way. This will definitely be an empty dict.